auto_update_tests.py

```diff
@@ -2,6 +2,8 @@
 
 import os, sys, subprocess, difflib
 
+os.environ['LD_LIBRARY_PATH'] = 'lib' # find our dynamic libraries
```

Comment:
Without static libs it means all free-standing uses need to set `LD_LIBRARY_PATH`, which is pretty annoying for users I think. Why not stick with static in that case? rpath is the common thing otherwise.